docs/4.2/ssh_azuread.md

```diff
@@ -0,0 +1,190 @@
+   
+6. Select Single Sign-on under Manage and choose SAML
+   ![Select SAML](img/azuread/azuread-6-selectsaml.png)
+   
+7. Select to edit  Basic SAML Configuration
+   ![Edit Basic SAML Configuration](img/azuread/azuread-7-editbasicsaml.png)
+   
+8. Put in the Entity ID and Reply URL the same proxy url https://proxy1:3080/v1/webapi/saml/acs
+   ![Put in Entity ID and Reply URL](img/azuread/azuread-8-entityandreplyurl.png)
+   
+9. Edit User Attributes & Claims
+
+a. Edit the Claim Name Change the name identifier format to Default Make sure the source attribute is user.userprincipalname
+   ![Confirm Name Identifier](img/azuread/azuread-9a-nameidentifier.png)
+   
+b. Add a group Claim to have user security groups groups
+   ![Put in Security group claim](img/azuread/azuread-9b-groupclaim.png)
+   
+10. On the SAM Signing Certificate selec to SAML Download the Federation Metadata XML.  
+   ![Download Federation Metadata XML](img/azuread/azuread-10-fedmeatadataxml.png)
+
+!!! warning "Important"  This a imporant document.  Treat this file as you would a password.
+
+## Create a SAML Connector
+
+Now, create a SAML connector [resource](admin-guide.md#resources).  Replace the <proxy address> with your teleport address and the group ids with the AzureAD group values.
+Write down this template as `azure-connector.yaml`:
+
+```yaml
+kind: saml
+version: v2
+metadata:
+  # the name of the connector
+  name: saml
+spec:
+  display: "Microsoft"
+  # acs is the Assertion Consumer Service URL. This should be the address of
+  # the Teleport proxy that your identity provider will communicate with.
+  acs: https://<proxy address>:3080/v1/webapi/saml/acs
+  attributes_to_roles:
+    - {name: "http://schemas.microsoft.com/ws/2008/06/identity/claims/groups", value: "<group id 930210...>", roles: ["admin"]}
+    - {name: "http://schemas.microsoft.com/ws/2008/06/identity/claims/groups", value: "<group id 93b110...>", roles: ["dev"]}
+  entity_descriptor: |
+    <federationmedata.xml contents>
+```
+
+Create the connector using `tctl` tool:
+
+```bsh
+$ tctl create azure-connector.yaml
+```
+
+## Create Teleport Roles
+
+We are going to create 2 roles:
+
+-  Privileged role admin who is able to login as root and is capable of administrating
+the cluster
+- Non-privileged dev
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: admin
+spec:
+  options:
+    max_session_ttl: 24h
+  allow:
+    logins: [root]
+    node_labels:
+      "*": "*"
+    rules:
+      - resources: ["*"]
+        verbs: ["*"]
+```
+
+Devs are only allowed to login to nodes labeled with `access: relaxed`
+Teleport label. Developers can log in as either `ubuntu` or a username that
+arrives in their assertions. Developers also do not have any rules needed to
+obtain admin access to Teleport.
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: dev
+spec:
+  options:
+    max_session_ttl: 24h
+  allow:
+    logins: [ "{% raw %}{{external.username}}{% endraw %}", ubuntu ]
```

Comment:
I have feeling that this is a longer string when coming from AD? See https://gravitational.com/teleport/docs/ssh_adfs/#create-teleport-roles

Something like `logins: [ '{{external."http://schemas.microsoft.com/ws/2008/06/identity/claims/windowsaccountname"}}', ubuntu ]`

Reply:
I provided instructions on how to make a shorter username with a claim